src/main/java/com/google/devtools/build/lib/analysis/skylark/StarlarkAttributeTransitionProvider.java

```diff
@@ -63,7 +66,8 @@ public StarlarkDefinedConfigTransition getStarlarkDefinedConfigTransitionForTest
   }
 
   @Override
-  public SplitTransition apply(AttributeMap attributeMap) {
+  public SplitTransition create(RuleTransitionData data) {
```

Comment:
What's the plan for `RuleTransitionData`? AFAIU it's just a slim wrapper around `AttributeMap` (plus three flags that are apparently about to be removed once the refactoring is complete and which are unused at HEAD and after this change) Is the plan to remove `RuleTransitionData` once those three bits can be removed?

Reply:
RuleTransitionData will eventually have more data: toolchain context, possibly a rule.

At the end of the migration, if it's possible to remove the wrapper, I will, but the main reason it is there is due to Bazel's package layout so I suspect it's there to stay.